### PR TITLE
iotools: reader for maccrad

### PR DESCRIPTION
**Pull request by dacoex**
a separate PR for MACC-RAD as requested in https://github.com/pvlib/pvlib-python/pull/274#issuecomment-264486119

- only 1 example file
- renamed to util

Discussion in #271

### CODE REVIEW

File: pvlib/iotools/maccrad.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+            lon = float(line.split(':')[1])
+    #    if line.startswith( "# Altitude"):
+        if "Altitude" in line:
+            alt_line = line
+            alt = float(alt_line.split(':')[1])
+    #        alt = float(line.split(':')[1])
+        if "Time reference" in line:
+            if "Universal time (UT)" in line:
+                tz_raw = 'UTC'
+            else:
```
Comment (wholmgren) [inline pvlib/iotools/maccrad.py]: Can these files ever have a time zone different from UTC or can they ever have no time zone information?

File: pvlib/iotools/maccrad.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    * coordinates
+    * altitude
+    
+    Retrieve timezone
+    """
+    if not name:
+        name = file_csv.split('.')[0]
+    
+    ## if file is on local drive
+    f = open(file_csv)
```
Comment (wholmgren) [inline pvlib/iotools/maccrad.py]: need to use a context manager to guarantee that the file will be closed.

File: pvlib/iotools/maccrad.py [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+            else:
+                logging.debug('No metadata on timezone found in input file')
+                logging.debug('Assuming UTC as default timezone')
+                tz_raw = 'UTC'
+
+    tz_loc = get_loc_latlon(lat, lon)
+    
+    
+    
+    location = Location(lat, lon, name=name, altitude=alt,
```
Comment (wholmgren) [inline pvlib/iotools/maccrad.py]: ``Location`` is part of the higher level API, so we don't want to force people into it in a low level reader. Instead, build and return a dict. You can add a constructor method to ``Location`` that consumes that dict.

File: pvlib/iotools/maccrad.py [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+                           tz=tz_loc)
+    
+    #XXX
+    metadata = None
```
Comment (wholmgren) [inline pvlib/iotools/maccrad.py]: what's this?

File: pvlib/iotools/maccrad.py [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+    * altitude
+    
+    Retrieve timezone
+    """
+    if not name:
+        name = file_csv.split('.')[0]
+    
+    ## if file is on local drive
+    f = open(file_csv)
+    for line in f:
```
Comment (wholmgren) [inline pvlib/iotools/maccrad.py]: Need to add a way to stop the iteration once we reach the end of the header.

File: pvlib/iotools/maccrad.py [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+        if "Title" in line:
+            title = line.split(':')[1].split('(')[0].strip()
+            name = title
```
Comment (wholmgren) [inline pvlib/iotools/maccrad.py]: I'm confused about what you want ``name`` to be set to. The title line contains a generic description of the data source.

File: pvlib/iotools/maccrad.py [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+    #    print (line)
+    #    if line.startswith( "# Latitude"):
+            lat_line = line
+            lat = float(lat_line.split(':')[1])
+    #        lat = float(line.split(':')[1])
+        if "Longitude" in line:
+    #    if line.startswith( "# Longitude"):
+            lon_line = line
+            lon = float(lon_line.split(':')[1])
+            lon = float(line.split(':')[1])
```
Comment (wholmgren) [inline pvlib/iotools/maccrad.py]: duplicate

File: pvlib/iotools/util.py [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+"""
+
+
+import pytz
+# How to get a time zone from a location using 
+# latitude and longitude coordinates?
+# http://stackoverflow.com/a/16086964
+# upstream: https://github.com/MrMinimal64/timezonefinder
+from timezonefinder import TimezoneFinder
```
Comment (wholmgren) [inline pvlib/iotools/util.py]: see comment here: https://github.com/pvlib/pvlib-python/pull/274#issuecomment-264486119

File: pvlib/iotools/util.py [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+    """Returns the timezone for a coordinate pair.
+    """
+    
+    
+    tf = TimezoneFinder()
+    tz = tf.timezone_at(lng=lon, lat=lat)
+    
+    return tz
+    
+def localise_df(df_notz, tz_source_str='UTC', tz_target_str=None):
```
Comment (wholmgren) [inline pvlib/iotools/util.py]: Why do we need this function? Seems more clear and robust to only call the pandas methods.

File: pvlib/iotools/maccrad.py [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+def read_maccrad_metadata(file_csv, name='maccrad'):
+    """
+    Read metadata from commented lines of the file
+    * coordinates
+    * altitude
+    
+    Retrieve timezone
+    """
+    if not name:
+        name = file_csv.split('.')[0]
```
Comment (wholmgren) [inline pvlib/iotools/maccrad.py]: This sounds fragile. Check out the ``os`` module for how to separate the file name from the full path. Next, are you trying to drop ``.csv`` from the name with the split? If so, do that explicitly.

File: pvlib/iotools/maccrad.py [hunk 11]
```diff
@@ -0,0 +1,155 @@
...
+    """
+    datetime_stamp = date.split('/')[0]
+
+    return datetime_stamp
+    
+#XXX read metadata / header
+
+def read_maccrad_metadata(file_csv, name='maccrad'):
+    """
+    Read metadata from commented lines of the file
```
Comment (wholmgren) [inline pvlib/iotools/maccrad.py]: numpydoc format

File: pvlib/iotools/maccrad.py (hunk 10 above)
Comment (mendomusic) [inline pvlib/iotools/maccrad.py]: >If so, do that explicitly.

What do you mean?

**Comment by wholmgren** [comment]
Still interested in the feature, but this PR would need to be reopened against the master branch.

**Pull request closed without merging**